### ISSUE TITLE
Linter throws the wrong error on en spaces (U+2002)

### ISSUE THREAD

**Issue by etjossem** [brigade/scss-lint#775]
In scss-lint 0.38.0, given the following variable declaration:

```scss
// Variations on $slate
$slate-60:      #7f90a0;
```

Scss-lint throws the following error:

`app/assets/stylesheets/style-guide/base/_colors.scss:44 [W] ColorVariable: Color literals like '#7f90a0' should only be used in variable declarations; they should be referred to via variable everywhere else.`

This error should never occur on variable declarations.

The error does not occur if the variable name is changed to the following, so it may have something to do with numbers in variable names:

```scss
// Variations on $slate
$slate-muted:      #7f90a0;
```

... but we also have `$blue-60: rgba($blue, .6);` and it passes, so this seems Very Weird.

**Comment by etjossem** [comment]
Update: this may be a dupe of https://github.com/brigade/scss-lint/pull/612

**Comment by etjossem** [comment]
This appears to be resolved by the current version 0.47.1.

**Comment by etjossem** [comment]
Hold on, coworker and I are looking into this. Seems to have something to do with unicode characters for the spaces in between variable name and hexcolor.

**Comment by etjossem** [comment]
Tracked it down. Using U+2002 ([En Space](http://www.fileformat.info/info/unicode/char/2002/index.htm)) anywhere in the definition will cause this error to throw, regardless of whether it's a variable definition or not. 

Copypastable: ` `

**Comment by sds** [comment]
After looking into this, this turns out to be a bug in the Sass parser.

If you include a non-standard space character on a line of Sass code as you have above, that entire line is parsed as a `Sass::Script::Tree::ListLiteral` instead of a `Sass::Tree::VariableNode`. This makes it impractical for SCSS-Lint to detect that we're in a variable declaration.

Please open this bug on the [Sass](https://github.com/sass/sass/issues/new) project. Thanks!

**Comment by etjossem** [comment]
Opened as https://github.com/sass/sass/issues/2126.
